index.d.ts

```diff
@@ -83,7 +83,8 @@ export declare interface Tracer extends opentracing.Tracer {
    * unless there is already an active span or `childOf` option. Note that this
    * option is deprecated and has been removed in version 4.0.
    */
-  trace<T> (name: string, fn: (span?: Span, fn?: (error?: Error) => any) => T): T;
+  trace<T> (name: string, fn: (span: Span) => T): T;
+  trace<T> (name: string, fn: (span: Span, done: (error?: Error) => string) => T): T;
```

Comment:
It was mentioned in https://github.com/DataDog/dd-trace-js/pull/2693 that return type of `done` callback should be `void`.

Reply:
Yep, missed that. Should be fixed now. 😅

Reply:
Looks like test.ts [needs](https://github.com/DataDog/dd-trace-js/actions/runs/7385905479/job/20091536262?pr=3912) to be updated too 😄

Reply:
Ah, right...that means it's a breaking change so I just removed the change to void. We'll have to make a not to switch that return value in the next major. 🤔

Reply:
Actually, looking at it more carefully, other changes are technically major too so I'll just mark the whole PR as major and try to land that in v5.